File: 1299. Replace Elements with Greatest Element on Right Side.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        overall_max = arr[-1]
        for i in range(len(arr) - 1, -1, -1):
            print(i)
            curr = arr[i]
            arr[i] = overall_max
            overall_max = max(curr, overall_max)
        arr[-1] = -1
        return arr

File: Arrays & Hashing/1299. Replace Elements with Greatest Element on Right Side.cpp
#include <iostream>
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<int> replaceElements(vector<int>& arr) {
        int n = arr.size();
        int curr_max = arr[n - 1];

        for(int i = n - 2; i >= 0; i--){
            int curr_elem = arr[i];
            arr[i] = curr_max;
            curr_max = max(curr_elem, curr_max);
        }
        arr[n - 1] = -1;
        
        return arr;
    }
};